Source/WebCore/rendering/svg/SVGRenderSupport.cpp

```diff
@@ -141,25 +141,33 @@ LayoutRepainter::CheckForRepaint SVGRenderSupport::checkForSVGRepaintDuringLayou
     return LayoutRepainter::CheckForRepaint::Yes;
 }
 
-auto SVGRenderSupport::computeContainerBoundingBoxes(const RenderElement& container, RepaintRectCalculation repaintRectCalculation) -> ContainerBoundingBoxes
+static bool hasValidBoundingBoxForContainer(const RenderObject& object)
 {
-    ContainerBoundingBoxes result;
+    if (auto* svgShape = dynamicDowncast<LegacyRenderSVGShape>(object))
+        return svgShape && !svgShape->isRenderingDisabled();
 
-    for (CheckedRef current : childrenOfType<RenderObject>(container)) {
-        if (current->isLegacyRenderSVGHiddenContainer())
-            continue;
+    if (auto* text = dynamicDowncast<RenderSVGText>(object))
+        return text && text->isObjectBoundingBoxValid();
```

Comment:
Why do we have to check `text` again in the return statement? Is not it enough to say:
```
   if (auto* text = dynamicDowncast<RenderSVGText>(object))
        return text->isObjectBoundingBoxValid();
```